feature_engine/timeseries/forecasting/lag_features.py

```diff
@@ -0,0 +1,313 @@
+    ) -> None:
+
+        if (
+            isinstance(periods, int)
+            and periods > 0
+            or isinstance(periods, list)
+            and all(isinstance(num, int) and num > 0 for num in periods)
+        ):
+            self.periods = periods
+        else:
+            raise ValueError(
+                "periods must be an integer or a list of positive integers. "
+                f"Got {periods} instead."
+            )
+
+        if missing_values not in ["raise", "ignore"]:
+            raise ValueError(
+                "missing_values takes only values 'raise' or 'ignore'. "
+                f"Got {missing_values} instead."
+            )
+
+        if not isinstance(drop_original, bool):
+            raise ValueError(
+                "drop_original takes only boolean values True and False. "
+                f"Got {drop_original} instead."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.freq = freq
+        self.missing_values = missing_values
+        self.drop_original = drop_original
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn parameters.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training dataset.
+
+        y: pandas Series, default=None
+            y is not needed in this transformer. You can pass None or y.
+        """
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find or check for numerical variables
+        self.variables_ = _find_or_check_numerical_variables(X, self.variables)
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        self.feature_names_in_ = X.columns.tolist()
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Adds lag features.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The data to transform.
+
+        Returns
+        -------
+        X_new: Pandas dataframe, shape = [n_samples, n_features + lag_features]
+            The dataframe with the original plus the new variables.
+        """
+        # Check method fit has been called
+        check_is_fitted(self)
+
+        # check if 'X' is a dataframe
+        X = _is_dataframe(X)
+
+        # Check if input data contains same number of columns as dataframe used to fit.
+        _check_input_matches_training_df(X, self.n_features_in_)
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        # if freq is not None, it overrides periods.
```

Comment:
Is the change to ensure that if a list is passed to both `freq` and `periods` that this method raises error happening in a different PR?

Reply:
We are not raising an error. The logic should just not use both and ensures that if freq is given, either list or string, it will override periods. And we mention that in the docstrings above.